roboteam_ai/src/utilities/Constants.cpp

```diff
@@ -0,0 +1,187 @@
+//
+// Created by mrlukasbos on 7-2-19.
+//
+
+#include "Constants.h"
+
+namespace rtt {
+namespace ai {
+
+// static variable initialization
+bool Constants::useGrSim = false;
+std::map<std::string, double> Constants::doubles;
+std::map<std::string, bool> Constants::bools;
+std::map<std::string, int> Constants::integers;
+std::map<std::string, QColor> Constants::colors;
+
+double Constants::getDouble(const std::string &name) {
+    if (doubles.find(name) != doubles.end() ) {
+        return doubles[name];
+    }
+    std::cerr << "Double constant not found: " << name.c_str() << std::endl;
+    return 0.0;
+}
+
+int Constants::getInt(const std::string &name) {
+    if (integers.find(name) != integers.end() ) {
+        return integers[name];
+    }
+    std::cerr << "Integer constant not found: " << name.c_str() << std::endl;
+    return 0;
+}
+
+bool Constants::getBool(const std::string &name) {
+    if (bools.find(name) != bools.end() ) {
+        return bools[name];
+    }
+    std::cerr << "Bool constant not found: " << name.c_str() << std::endl;
+    return false;
+}
+
+QColor Constants::getColor(const std::string &name) {
+    if (colors.find(name) != colors.end() ) {
+        return colors[name];
+    }
+    std::cerr << "Color constant not found: " << name.c_str() << std::endl;
+    return { 0, 0, 0, 0 }; // transparent
+}
+
+void Constants::init() {
+    ros::NodeHandle nh;
+    std::string robotOutputTarget;
+    nh.getParam("robot_output_target", robotOutputTarget);
+    useGrSim = robotOutputTarget != "serial"; // use the grSim as default; only serial is specifically defined
+
+    bools["SHOW_LONGEST_TICK"]              = true;
```

Comment:
Will this allow CLion to auto-finish constant names for you. That's what I liked about the old implementation of constants. Besides that I really like this!